src/nupic/algorithms/Connections.cpp

```diff
@@ -144,6 +144,10 @@ void Connections::updateSynapsePermanence(const Synapse& synapse,
   const Cell& cell = segment.cell;
 
   cells_[cell.idx].segments[segment.idx].synapses[synapse.idx].permanence = permanence;
+
+  if (permanence == 0) {
```

Comment:
permance is a floating point number so better do do `permanence < 0.00001` or whatever an appropriate epsilon is